src/layout/SigningStatusPanel/PanelAwaitingCurrentUserSignature.tsx

```diff
@@ -33,32 +39,63 @@ export function AwaitingCurrentUserSignaturePanel({
   const selectedLanguage = useCurrentLanguage();
   const queryClient = useQueryClient();
   const textResourceBindings = useNodeItem(node, (i) => i.textResourceBindings);
+  const { langAsString } = useLanguage();
 
   const title = textResourceBindings?.awaitingSignaturePanelTitle ?? 'signing.awaiting_signature_panel_title';
   const checkboxLabel = textResourceBindings?.checkboxLabel ?? 'signing.checkbox_label';
   const checkboxDescription = textResourceBindings?.checkboxDescription;
   const signingButtonText = textResourceBindings?.signingButton ?? 'signing.sign_button';
 
+  const {
+    data: authorizedOrganisationDetails,
+    isLoading: isApiLoading,
+    error: apiError,
+  } = useQuery(authorizedOrganisationDetailsQuery(instanceOwnerPartyId!, instanceGuid!));
+
   const {
     mutate: handleSign,
     error,
     isSuccess,
     isPending,
   } = useMutation({
-    mutationFn: async () => {
+    mutationFn: async (onBehalfOf: string) => {
       if (instanceOwnerPartyId && instanceGuid) {
-        return doPerformAction(instanceOwnerPartyId, instanceGuid, { action: 'sign' }, selectedLanguage, queryClient);
+        return doPerformAction(
+          instanceOwnerPartyId,
+          instanceGuid,
+          { action: 'sign', onBehalfOf },
```

Comment:
Blir det _bittelit_ mer riktig med `onBehalfOf` som `null` i stedet for tom streng når det ikke er på vegne av noen enn seg selv? 😊